### PR TITLE
Match weather variables

### PR DESCRIPTION
**Pull request by jessicaguo**
Daymet (daily) and NOAA (hourly) weather data harmonized in terms of timing and units. Now organized to output the same variables: siteID, date, radiation (MJ/m2/day), max_temp (C), min_temp (C), precip (mm/day) , and vpd (kPa, NOAA only)
VPD could not be calculated for daily Daymet data due to non-linearity between temp and SVP. 

There were 31 weather ensembles provided. Current script summarizes as the median.

### CODE REVIEW

File: pheno_images/curate_weather_forecasts.R [hunk 1]
```diff
@@ -1,7 +1,44 @@
...
+         precip = ud.convert(precipitation_flux, "s^-1", "d^-1"), #kg per m2 is equivalent to mm
+         vpd = RHtoVPD(RH = relative_humidity, TdegC = airtemp_C)) %>%
+  group_by(siteID, time) %>%
+  summarize(radiation = median(surface_downwelling_shortwave_flux_in_air),
+            airtemp_C = median(airtemp_C),
+            precip = median(precip),
+            vpd = median(vpd),
+            rad_Mj_hr = ud.convert(radiation*60*60, "joule", "megajoule")) %>%
+  ungroup() %>%
+  mutate(date = as.Date(substr(time, 1, 19)))
```
Comment (KristinaRiemer) [inline pheno_images/curate_weather_forecasts.R:31]: There's something going on with this conversion to date, I'm not sure what? The `time` column at that point is incrementing values starting at 0 per hour.

File: pheno_images/curate_weather_forecasts.R [hunk 2]
```diff
@@ -1,7 +1,44 @@
...
+
+# Then, summarize to daily. Note that precipitation is cumulative, so take max rather than sum
+daily <- hourly %>%
+  group_by(siteID, date) %>%
+  summarize(radiation = sum(rad_Mj_hr),
+            max_temp = max(airtemp_C),
+            min_temp = min(airtemp_C),
+            precip = max(precip),
```
Comment (KristinaRiemer) [inline pheno_images/curate_weather_forecasts.R:39]: ~~I think I need a better explanation of the precipitation flux conversion here to understand. The unit is converted to daily values per hour, and then the max value is selected so it's the highest rate of rain per day. But wouldn't the final precip units be kg/m2, while the Daymet precip is mm?~~ TIL kg/m2 == mm!

File: pheno_images/curate_weather.R [hunk 3]
```diff
@@ -24,11 +26,26 @@ for(site in 1:nrow(efi_sites)){
...
          date = as.Date(yday, origin = origin_date)) %>% 
-  select(-year, -yday, -origin_year, -origin_date)
+  select(-year, -yday, -origin_year, -origin_date) %>%
+  rename(siteID = site,
+         max_temp = tmax..deg.c.,
+         min_temp = tmin..deg.c.,
+         precip = prcp..mm.day.) %>%
+  relocate(siteID, date) %>%
+  mutate(radiation = ud.convert(dayl..s.*srad..W.m.2., "joule", "megajoule")) %>%
+         # SVP = esat(max_temp),
```
Comment (KristinaRiemer) [inline pheno_images/curate_weather.R:41]:
```suggestion
```

File: pheno_images/curate_weather.R [hunk 4]
```diff
@@ -24,11 +26,26 @@ for(site in 1:nrow(efi_sites)){
...
+         # vpd = ud.convert(SVP - vp..Pa., "Pa", "kPa"))
```
Comment (KristinaRiemer) [inline pheno_images/curate_weather.R:42]:
```suggestion
```

File: pheno_images/curate_weather.R [hunk 5]
```diff
@@ -24,11 +26,26 @@ for(site in 1:nrow(efi_sites)){
...
+  select(siteID, date, radiation, max_temp, min_temp, precip) %>%
+  arrange(siteID)
+
+#### Write out only weather data file ####
+write.csv(clean_daymet, file = "Daymet_weather.csv", row.names = F)
+
 
 ###########Join weather data to GCC data###########
 pheno_images <- read.csv("pheno_images/targets_gcc.csv") %>%
```
Comment (KristinaRiemer) [inline pheno_images/curate_weather.R:51]: Due to changes in the final `clean_daymet` dataframe, the joining with the GCC dataset needs to change as below. 
```suggestion
pheno_images <- read.csv("pheno_images/targets_gcc.csv") %>% 
  mutate(time = as.Date(time))

gcc_weather <- left_join(pheno_images, clean_daymet, 
                         by = c("siteID" = "siteID", "time" = "date"))

```

**Review by KristinaRiemer — CHANGES_REQUESTED** [review]

File: pheno_images/curate_weather_forecasts.R (hunk 1 above)
Comment (jessicaguo) [reply pheno_images/curate_weather_forecasts.R:31]: I'm not sure what the issue is. Just ran it again and didn't get an error. Could you please clarify?

File: pheno_images/curate_weather_forecasts.R (hunk 2 above)
Comment (jessicaguo) [reply pheno_images/curate_weather_forecasts.R:39]: Haha, it's confusing for sure! Due to water having standard volume/mass conversions.

File: pheno_images/curate_weather.R (hunk 3 above)
Comment (jessicaguo) [reply pheno_images/curate_weather.R:41]: Done

File: pheno_images/curate_weather.R (hunk 4 above)
Comment (jessicaguo) [reply pheno_images/curate_weather.R:42]: Done

**Review by KristinaRiemer — APPROVED** [review]

**Pull request merged**